compiler/rustc_hir_analysis/src/lib.rs

```diff
@@ -494,6 +494,18 @@ pub fn check_crate(tcx: TyCtxt<'_>) -> Result<(), ErrorGuaranteed> {
         tcx.hir().for_each_module(|module| tcx.ensure().check_mod_item_types(module))
     });
 
+    // FIXME: Parallelize other iterators(e.g. `body_owners` in `check_unused`) instead
+    // when we implement creating `DefId`s for anon constants during their parents' typeck.
+    // Since doing so will produce query cycle errors at current.
+    if cfg!(parallel_compiler) {
```

Comment:
Actually it doesn't change the logic, because `check_unused` will call all `typeck` and fetch the `used_trait_impl` information in the results. I moved this part into `check_unused` to make it look more reasonable.

Reply:
This code really belongs in `check_crate`, and not in `check_unused`. My suggestion was just to remove the `if cfg!(parallel_compiler)`, and run this unconditionally.

Reply:
This reverts part of #108118. I agree this is reasonable.

Reply:
I removed `cfg!(parallel_compiler)` and currented some ui tests